### PR TITLE
feat: add batch request

### PR DESCRIPTION
**Pull request by jesrypandawa**
- fix validation for url config
- add batch request

### CODE REVIEW

File: src/main/java/io/odpf/depot/http/request/BatchRequest.java [hunk 1]
```diff
@@ -1,14 +1,62 @@
...
+            });
+            Map<String, String> requestHeaders = headerBuilder.build();
+            URI requestUrl = uriBuilder.build(Collections.emptyMap());
+            HttpEntityEnclosingRequestBase request = RequestMethodFactory.create(requestUrl, httpMethod);
+            requestHeaders.forEach(request::addHeader);
+            request.setEntity(buildEntity(payloads.toString()));
+            record = new HttpRequestRecord((long) 0, null, true, request);
+            return Collections.singletonList(record);
+        } catch (Exception e) {
+            Map<String, Object> metadata = messages.get(0).getMetadata();
```
Comment (lavkesh) [inline src/main/java/io/odpf/depot/http/request/BatchRequest.java:55]: can we define what kind of exceptions are being caught?

File: src/main/java/io/odpf/depot/http/request/BatchRequest.java [hunk 2]
```diff
@@ -1,14 +1,62 @@
...
-        return null;
+        List<String> payloads = new ArrayList<>();
+        HttpRequestRecord record;
+        try {
+            IntStream.range(0, messages.size()).forEach(index -> {
+                String body = requestBody.build(messages.get(index));
+                if (body != null) {
+                    payloads.add(body);
+                }
+            });
```
Comment (lavkesh) [inline src/main/java/io/odpf/depot/http/request/BatchRequest.java:46]: List<String> payloads = messages.stream().map(requestBody::build).collect(Collectors.toList());

File: src/main/java/io/odpf/depot/http/request/Request.java [hunk 3]
```diff
@@ -1,11 +1,22 @@
...
 public interface Request {
 
     List<HttpRequestRecord> createRecords(List<OdpfMessage> messages);
+
+    default StringEntity buildEntity(String stringBody) {
+        return new StringEntity(stringBody, ContentType.APPLICATION_JSON);
+    }
+
+    default HttpRequestRecord createAndLogErrorRecord(ErrorInfo errorInfo, int index) {
+        return new HttpRequestRecord((long) index, errorInfo, false, null);
```
Comment (lavkesh) [inline src/main/java/io/odpf/depot/http/request/Request.java:20]: this should only be createErrorRecord(). it's not doing logging.

File: src/main/java/io/odpf/depot/http/request/BatchRequest.java (hunk 2 above)
Comment (lavkesh) [reply src/main/java/io/odpf/depot/http/request/BatchRequest.java:46]: `
     try {
            Map<String, String> requestHeaders = headerBuilder.build();
            URI requestUrl = uriBuilder.build(Collections.emptyMap());
            HttpEntityEnclosingRequestBase request = RequestMethodFactory.create(requestUrl, httpMethod);
            requestHeaders.forEach(request::addHeader);
            String payload = messages.stream().map(requestBody::build).collect(Collectors.toList()).toString();
            request.setEntity(buildEntity(payload));
            record = new HttpRequestRecord((long) 0, null, true, request);
            return Collections.singletonList(record);
        }
`

File: src/main/java/io/odpf/depot/http/request/BatchRequest.java (hunk 1 above)
Comment (jesrypandawa) [reply src/main/java/io/odpf/depot/http/request/BatchRequest.java:55]: we want to handle all the possible exceptions in another story right?

File: src/main/java/io/odpf/depot/http/request/BatchRequest.java (hunk 2 above)
Comment (jesrypandawa) [reply src/main/java/io/odpf/depot/http/request/BatchRequest.java:46]: how we can handle null message? if its null then null string will be included in the body.

File: src/main/java/io/odpf/depot/http/request/Request.java (hunk 3 above)
Comment (jesrypandawa) [reply src/main/java/io/odpf/depot/http/request/Request.java:20]: ack

**Pull request closed without merging**